File: Problems/Palindrom Number/PalindromNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        String xString = Integer.toString(x);
        int xStringLength = Integer.toString(x).length();
        int j = xStringLength-1;
        for (int i = 0; i < xStringLength; i++){
            if (xString.charAt(i) != xString.charAt(j)){
                return false;
            }
            j--;
        }
        return true;
    }
}

File: Problems/Palindrom Number/PalindromNumber.c
bool isPalindrome(int x) {
    
    if (x < 0) {
        return false;
    }
    else if (x == 0) {
        return true;
    }

    int numDigits = 0;
    int n = x;

    while (n > 0) {
        n = n / 10;
        numDigits++;
    }

    int nums[numDigits];

    for (int i = 0; i < numDigits; i++) {
        nums[i] = x % 10;
        x /= 10;
    }

    for (int i = 0; i < numDigits/2; i++) {
        if (nums[i] != nums[numDigits-1-i]){
            return false;
        }
    }

    return true;

}
